crosschain-resolver/contracts/L1Resolver.sol

```diff
@@ -273,6 +267,18 @@ contract L1Resolver is EVMFetchTarget, ITargetResolver, IMetadataResolver, IExte
             interfaceId == type(IExtendedResolver).interfaceId ||
             interfaceId == type(ITargetResolver).interfaceId ||
             interfaceId == type(IMetadataResolver).interfaceId ||
+            interfaceId == type(IResolverSetter).interfaceId ||
             super.supportsInterface(interfaceId);
     }
+
+    function convertCoinTypeToEVMChainId(uint256 coinType) private pure returns (uint256) {
+        return (0x7fffffff & coinType) >> 0;
+    }
+
+    function _writeDeferral(address target) internal view {
+        revert StorageHandledByL2(
+            convertCoinTypeToEVMChainId(l2ResolverCoinType),
```

Comment:
Why not just store and use a chainID directly?

Also, this doesn't seem to be set anywhere?